src/examples/2.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Adapted from: https://stackoverflow.com/questions/32761930/
+ *
+ * Returns the integer value of a string containing an integer.
+ *
+ * Note: A previous routine (not included here) ensures:
+ *  - `columnSortSetting` always contains a finite integer
+ *
+ * @param columnSortSetting Finite integer value
+ * @returns The integer if valid; otherwise, null
+ */
+export function getSortSetting(columnSortSetting: number): number | null {
+  return columnSortSetting || null;
```

Comment:
This example relies on the same "trick" as the 1.ts—`0` evaluates as `false`.

Reply:
Right, it's a pretty common mistake.  If we keep only one of the two, I would pick `2.ts` as it is the simpler example.